File: src/main/java/net/bms/genera/items/ItemConnlaRing.java
package net.bms.genera.items;

import baubles.api.BaubleType;
import baubles.api.IBauble;
import net.bms.genera.Genera;
import net.bms.genera.util.Constants;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

public class ItemConnlaRing extends Item implements IBauble {
    public ItemConnlaRing() {
        setCreativeTab(Genera.TabGenera);
        setMaxStackSize(1);
        setRegistryName(Constants.MODID, "ring_of_connla");
        setUnlocalizedName("ring_of_connla");
    }

    @Override
    public BaubleType getBaubleType(ItemStack itemstack) {
        return BaubleType.RING;
    }

    @Override
    public void onWornTick(ItemStack itemstack, EntityLivingBase player) {

    }

    @Override
    public void onEquipped(ItemStack itemstack, EntityLivingBase player) {

    }

    @Override
    public void onUnequipped(ItemStack itemstack, EntityLivingBase player) {

    }

    @Override
    public boolean canEquip(ItemStack itemstack, EntityLivingBase player) {
        return false;
    }

    @Override
    public boolean canUnequip(ItemStack itemstack, EntityLivingBase player) {
        return false;
    }

    @Override
    public boolean willAutoSync(ItemStack itemstack, EntityLivingBase player) {
        return false;
    }
}

File: src/main/kotlin/net/bms/genera/item/ItemConnlaRing.kt
package net.bms.genera.item

import baubles.api.BaubleType
import baubles.api.IBauble
import net.bms.genera.Genera
import net.minecraft.entity.EntityLivingBase
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.potion.Potion
import net.minecraft.potion.PotionEffect
import net.minecraftforge.fml.common.Optional

@Optional.Interface(iface = "baubles.api.IBauble", modid = "baubles", striprefs = true)
class ItemConnlaRing: Item(), IBauble {
    init {
        creativeTab = Genera.TabGenera
        setMaxStackSize(1)
        setRegistryName(Genera.MODID, "ring_of_connla")
        unlocalizedName = "ring_of_connla"
    }

    override fun getBaubleType(itemstack: ItemStack): BaubleType {
        return BaubleType.RING
    }

    override fun onWornTick(itemstack: ItemStack?, player: EntityLivingBase?) {
        val strength = Potion.getPotionById(5)
        if (strength != null && !player!!.isPotionActive(strength)) {
            player.addPotionEffect(PotionEffect(strength, 300, 2, false, false))
        }
    }

    override fun onEquipped(itemstack: ItemStack?, player: EntityLivingBase?) {

    }

    override fun onUnequipped(itemstack: ItemStack?, player: EntityLivingBase?) {

    }

    override fun canEquip(itemstack: ItemStack?, player: EntityLivingBase?): Boolean {
        return true
    }

    override fun canUnequip(itemstack: ItemStack?, player: EntityLivingBase?): Boolean {
        return true
    }

    override fun willAutoSync(itemstack: ItemStack?, player: EntityLivingBase?): Boolean {
        return false
    }
}
